From 5bb35197b0b6e6c1ad9232d7129cd89bc0a23cf2 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Wed, 3 Jun 2026 21:10:21 +0800
Subject: [PATCH 2/4] ActionReceiver: Inject RECEIVER_NOT_EXPORTED when running
 registerReceiver()

E AndroidRuntime: java.lang.SecurityException: com.oneplus.gallery: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
---
 .../uikit/broadcast/bus/ActionReceiver.smali           | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/smali_classes2/com/oplus/gallery/foundation/uikit/broadcast/bus/ActionReceiver.smali b/smali_classes2/com/oplus/gallery/foundation/uikit/broadcast/bus/ActionReceiver.smali
index 57b1dd1ab..2576e87d6 100644
--- a/smali_classes2/com/oplus/gallery/foundation/uikit/broadcast/bus/ActionReceiver.smali
+++ b/smali_classes2/com/oplus/gallery/foundation/uikit/broadcast/bus/ActionReceiver.smali
@@ -189,6 +189,16 @@
     .line 61
     iget v7, v0, Lcom/oplus/aiunit/vision/fx0;->c:I
 
+    # --- START PATCH: conditionally add RECEIVER_NOT_EXPORTED (0x4) ---
+    and-int/lit8 v0, v7, 0x6        # 0x6 = 0x2 | 0x4 (both export flags)
+    if-eqz v0, :cond_no_or
+    # Already has EXPORTED or NOT_EXPORTED, keep as is
+    goto :cond_skip_or
+    :cond_no_or
+    or-int/lit8 v7, v7, 0x4
+    :cond_skip_or
+    # --- END PATCH ---
+
     move-object v3, p0
 
     .line 64
-- 
2.55.0

